jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/util/ArtifactUtil.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2019 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at:
+ *
+ *     https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.jkube.kit.common.util;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.math.BigInteger;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+import org.eclipse.jkube.kit.common.PrefixedLogger;
+
+public class ArtifactUtil {
+  protected static final String CHECKSUM_FILENAME = ".jkube-checksum";
+
+  private ArtifactUtil() {
+
+  }
+
+  public static boolean isArtifactSameAsPrevious(File sourceFile, Path checksumPath, PrefixedLogger log)
+      throws IOException, NoSuchAlgorithmException {
+    byte[] data = Files.readAllBytes(sourceFile.toPath());
```

Comment:
Could you move the whole checksum calculation to a dedicated function?
`getCurrentOutputArtifactChecksum`